pkg/operator/controller/controller_dns_configmap_test.go

```diff
@@ -1141,6 +1141,76 @@ func Test_coreDNSCache(t *testing.T) {
 	}
 }
 
+func TestDesiredDNSConfigmapDNSNameResolverEnabled(t *testing.T) {
+	testCases := []struct {
+		name             string
+		dns              *operatorv1.DNS
+		namespaces       []string
+		expectedCoreFile string
+		expectedError    error
+	}{
+		{
+			name: "Check if Corefile is rendered correctly with ocp_dnsnameresolver plugin",
+			dns: &operatorv1.DNS{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: DefaultDNSController,
+				},
+				Spec: operatorv1.DNSSpec{
+					Servers: []operatorv1.Server{
+						{
+							Name:  "foo",
+							Zones: []string{"foo.com"},
+							ForwardPlugin: operatorv1.ForwardPlugin{
+								Upstreams: []string{"1.1.1.1", "2.2.2.2:5353"},
+								Policy:    operatorv1.RoundRobinForwardingPolicy,
+							},
+						},
+						{
+							Name:  "bar",
+							Zones: []string{"bar.com", "example.com"},
+							ForwardPlugin: operatorv1.ForwardPlugin{
+								Upstreams: []string{"3.3.3.3"},
+								Policy:    operatorv1.RandomForwardingPolicy,
+							},
+						},
+					},
+				},
+			},
+			namespaces:       []string{"openshift-ovn-kubernetes"},
+			expectedCoreFile: mustLoadTestFile(t, "default_corefile_ocp_dnsnameresolver"),
+		},
+		{
+			name: "Check if Corefile is rendered correctly with ocp_dnsnameresolver plugin with multiple namespaces",
+			dns: &operatorv1.DNS{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: DefaultDNSController,
+				},
+				Spec: operatorv1.DNSSpec{},
+			},
+			namespaces:       []string{"foo", "bar", "foobar"},
+			expectedCoreFile: mustLoadTestFile(t, "default_corefile_ocp_dnsnameresolver_with_multiple_namespaces"),
+		},
+	}
+
+	clusterDomain := "cluster.local"
+	cmMap := make(map[string]string)
+	cmMap["cacerts"] = "ca-cacerts-2"
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if cm, err := desiredDNSConfigMap(tc.dns, clusterDomain, cmMap, true, tc.namespaces); err != nil {
+				if !errors.Is(err, tc.expectedError) {
+					t.Errorf("Unexpected error : %v", err)
+				}
+			} else if tc.expectedError != nil {
+				t.Errorf("Error %v was expected", tc.expectedError)
+			} else if diff := cmp.Diff(cm.Data["Corefile"], tc.expectedCoreFile); diff != "" {
```

Comment:
Would be nice to dry-run the corefile to check the syntax. But this can be an item for the future.